homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -368,6 +370,16 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
             );
         }
         SwarmEvent::IncomingConnection { .. } => {}
+        SwarmEvent::ConnectionEstablished {
+            peer_id, endpoint, ..
+        } => {
+            // add peer to connected peers list
+            event_handler.connected_peers.insert(peer_id, endpoint);
```

Comment:
I feel like we should add trusted nodes as explicit peers for gossipsub and remove them appropriately, right? I know we're implementing rendezvous too, so maybe that's redundant?, but trusted nodes discovered should be made explicit, right @mriise?

Reply:
the major issue is that explicit peers aren't counted towards the current mesh count of a topic. If all of the connected nodes are explicit then gossip won't let you send messages (since there is no-one counted in the mesh!).  That grows the minimum number of nodes in a network (using gossip) from just 2 up to explicit nodes + 1 non-explicit (since explicit peering agreements are supposed to be reciprocal).

Reply:
@mriise, I totally understand how they are not counted in the mesh, and the need for facilitating smaller networks is a great point. Nonetheless, it feels a bit odd to fully skip explicit peering agreement(s) altogether. Looking at [fleek's impl](https://github.com/fleek-network/ursa/blob/main/crates/ursa-network/src/service.rs#L406), they add explicit nodes for identity events that are in the same peer network, using the same protocol name for example. Nothing that prevents this iteration from going in, but something to still think about.

Reply:
I think there is a place for explicit nodes, but for the moment we should be fine to ignore since it is more of an optimization.

Reply:
yep, def can for now.